tron/serialize/runstate/statemanager.py

```diff
@@ -201,6 +204,22 @@ def _save_from_buffer(self):
                 log.warning(msg)
                 raise PersistenceStoreError(msg)
 
+    def _check_consistency(self, state):
+        """ Consistency checks on serialized state to ensure no corruption.
+
+        :param state: A dict of serialized state, like that returned by
+            self.restore
+        """
+        log.info("Running consistency checks")
+        try:
+            # We attempt to save all the state. A failure indicates corruption
+            for type_enum, items in state.items():
+                for name, state_data in items.items():
+                    self.save(type_enum, name, state_data)
+        except PersistenceStoreError as e:
+            log.error("Failed consistency check: Could not save serialized state")
+            raise e
```

Comment:
So, Tron would crash if the error is raised here, right?

Reply:
Yes, which is exactly what we want. If it fails the consistency check on start up, we should fail immediately.